File: packages/zod/src/audit.ts
import { z } from "zod";

export const ZAuditDecisionStatus = z.enum(["approved", "flagged", "rejected"]);

export const ZAdminClaimOverrideRequest = z.object({
  decision: ZAuditDecisionStatus,
  reason: z.string().trim().min(10).max(1000),
});

export const ZAuditResponse = z.object({
  id: z.string().uuid(),
  claimId: z.string().uuid(),
  decision: ZAuditDecisionStatus,
  reason: z.string(),
  citedPolicyText: z.string().nullable().optional(),
  confidence: z.number(),
  aiModel: z.string(),
  deterministicRule: z.string().nullable().optional(),
  overriddenBy: z.string().nullable().optional(),
  overrideReason: z.string().nullable().optional(),
  createdAt: z.string().datetime(),
});

export type AuditDecisionStatus = z.infer<typeof ZAuditDecisionStatus>;
export type AdminClaimOverrideRequest = z.infer<
  typeof ZAdminClaimOverrideRequest
>;
export type AuditResponse = z.infer<typeof ZAuditResponse>;

File: apps/backend/internal/model/audit.go
package model

import (
	"time"

	"github.com/Akshay2642005/expense-auditor/internal/errs"
	"github.com/google/uuid"
)

type AuditDecisionStatus string

const (
	AuditDecisionApproved AuditDecisionStatus = "approved"
	AuditDecisionFlagged  AuditDecisionStatus = "flagged"
	AuditDecisionRejected AuditDecisionStatus = "rejected"
)

type AuditDecision struct {
	ID                uuid.UUID           `json:"id"                          db:"id"`
	ClaimID           uuid.UUID           `json:"claimId"                     db:"claim_id"`
	Decision          AuditDecisionStatus `json:"decision"                    db:"decision"`
	Reason            string              `json:"reason"                      db:"reason"`
	CitedPolicyText   *string             `json:"citedPolicyText,omitempty"   db:"cited_policy_text"`
	Confidence        float64             `json:"confidence"                  db:"confidence"`
	AIModel           string              `json:"aiModel"                     db:"ai_model"`
	DeterministicRule *string             `json:"deterministicRule,omitempty" db:"deterministic_rule"`
	OverriddenBy      *string             `json:"overriddenBy,omitempty"      db:"overridden_by"`
	OverrideReason    *string             `json:"overrideReason,omitempty"    db:"override_reason"`
	RawModelOutput    *string             `json:"rawModelOutput,omitempty"    db:"raw_model_output"`
	CreatedAt         time.Time           `json:"createdAt"                   db:"created_at"`
}

type GetClaimAuditRequest struct {
	ID string `param:"id"`
}

func (r *GetClaimAuditRequest) Validate() error {
	if r.ID == "" {
		return errs.NewBadRequestError("id is required", true, nil, nil, nil)
	}
	if _, err := uuid.Parse(r.ID); err != nil {
		return errs.NewBadRequestError("id must be a valid UUID", true, nil, nil, nil)
	}
	return nil
}
